app/models/miq_expression.rb

```diff
@@ -684,7 +684,7 @@ def _to_sql(exp, tz)
         tag = exp[operator]["value"]
         klass = klass.constantize
         ids = klass.find_tagged_with(:any => tag, :ns => ns).pluck(:id)
-        clause = ActiveSupport::Deprecation.silence { "(#{klass.send(:sanitize_sql_for_conditions, :id => ids)})" }
+        clause = klass.send(:sanitize_sql_for_conditions, ["#{model.table_name}.id IN (?)", ids])
```

Comment:
There is the desire to go this way.
The new `sanitize_sql_for_conditions` does not support hashes.
So we first need to get away from legacy finders and probably use scopes instead.